### ISSUE TITLE
USTP: Legacy data needs to be imported once

### ISSUE THREAD

**Issue by urapadmin** [arch-kiosk/arch-kiosk-office#1789]
Conversation between @Tfranconi and @urapadmin  from July 2022:

Today is really the first day that I've gotten to sit down with my "finds director" hat on and think about material entry, and I am wondering about how best to go about entering the past 8 years of ceramic collection and documentation into the database. We have ~1700 contexts of pottery that clock over 40,000 sherds, so I'm wondering if there is a way to import csv files or something similar into the database rather than go through and enter them all by hand. I see clear value in going over everything again manually, but realistically I do not have that kind of time, so any kind of import would be a huge help.

Just to jog your memory, I attach the 2018 ceramic quantification spreadsheets (one by sherd by context and one by total sherds and weight) so that you can see the state of the data.

- #1811 is responsible for updating the weight and count in collected material after synchronization together with 
- #1809

**Comment by urapadmin** [comment]
There is no automatic import that I can imagine here. The information is too individually structured. Just two examples: Context numbers are stored in the sheet's title instead of in the data and I do see several incoherencies in naming contexts because Excel interpreted the context numbers as numeric values. Such are the merits of Excel. But importing data from pivot tables in general is a very individual thing to do. So I would have to tailor a semi-manual import for your particular project that we (presumably I myself) use once to get all the current legacy data in and then in the future you would only use kiosk to input data. You or a student would have to go through the Excel sheets and fix incoherent inputs. That is feasible and I agree that a manual input of that number of sherds would be insane. We can find a better solution. I have a few questions, though:
1. Is it right that the 2018_pottery_quantification_july_august.xlsx would be the raw data to import while the 2018_pottery_totals_july_august would be rather the result, so a report kiosk would have to produce?

2. A real issue for the import will be that those legacy contexts (and the collected material) will have to be created, too. Or do they already exist?  If not, they would have to be created automatically and that would beg the question: what should they look like? What data should be put in?

Since all that will be real work for all of us the most important question is: To what end would we do it? What is the purpose of the spreadsheets? Are they covering reporting obligations for the government or are you using them for analysis? Do you really need the sherd counts in the database or could you for instance just input new sherd data into kiosk and it would generate the same Excel sheets you are used to so that in the end you would keep that part of your project's data in Excel?

**Comment by urapadmin** [comment]
Lutz: 
For the ceramic data, this is the form it is is now, but I could change it to anything that might work. Could you imagine a format or table or something where it WOULD be possible? If so, what does that look like? For your questions:

1.) Yes, that's right
2.) Some of the legacy contexts are created, many, particularly older ones, do not. Our goal should be to have all of the legacy contexts entered, otherwise this is never going to work very well, so creating them should contain the same info as the sheets, ideally. 

The purpose of the spreadsheets is for me to study and publish the pottery, and also to be able to look up a context and come up with a datable element if it exists. When I write the ceramics volume, though, the synthetic overview of the material will be drawn from these excel tables. 

For the time being, I will enter new data from this year and keep the paper/excel records for a future project. But, while I'm here and thinking about it, if you have any thoughts on how to make it importable, now is a good time to have that discussion.

**Comment by urapadmin** [comment]
see also #1809 which takes care of duplicate category combinations per context/cm

**Comment by Tfranconi** [comment]
This sounds like a totally reasonable solution to me!

---

Tyler Franconi
Visiting Assistant Professor of Archaeology
Joukowsky Institute for Archaeology & the Ancient World
Brown University
Box 1837, 60 George Street, Providence, RI  02912, USA
***@***.***

**Comment by urapadmin** [comment]
This is what a typical 2014 ceramic sheet leads to in my import:

```
WARNING: Sheet 345: Cell H1 has superfluous column header mni -> ignored.
ERROR:   Sheet 345: Cell A2 has unknown value coursewares. -> sheet skipped
ERROR:   Sheet 345: Cell A9 has unknown value african ts. -> sheet skipped
ERROR:   Sheet 345: Cell A10 has unknown value post roman glazed. -> sheet skipped
ERROR:   Sheet 345: Cell A14 has unknown value rooftiles. -> sheet skipped
ERROR:   Sheet 345: Cell A15 has unknown value bricks. -> sheet skipped
WARNING: Sheet 345: Missing rows coarseware,stone,invetriata,maiolica,dolia,african rs. -> ignored.
ERROR:   Sheet 345 cannot be imported.
```
- [ ] I suspect "mni" can be ignored as only a few of those sheets actually have it
- [ ] Courseware I would translate into Coarseware

- what about the other uknown values?
- what about the missing values except from coarseware?

**Comment by urapadmin** [comment]
By the way: Don't got through the pain of changing these things in your Excel files. I can correct and transform values during the import. That way I can make sure that each instance is automatically consistently transformed.

**Comment by Tfranconi** [comment]
Hi Lutz,

Sorry for the slow response, but hey look--those inadequacies I mentioned!
I'm sorry I didn't take more time to clean these sheets, some of those
things would have been obvious to remove or repair. To respond in turn:

WARNING: Sheet 345: Cell H1 has superfluous column header mni -> ignored.
*This is the best thing, and probably only for this year.*

ERROR:   Sheet 345: Cell A2 has unknown value coursewares. -> sheet skipped
*Yup, coarsewares.*

ERROR:   Sheet 345: Cell A9 has unknown value african ts. -> sheet skipped
*Should be ARS or African Red Slip (whatever the name is in the database,
there should be two african categories--red slip and cookware.) African ts
would be the same as ARS or African Red Slip.*

ERROR:   Sheet 345: Cell A10 has unknown value post roman glazed. -> sheet
skipped
*This should be classed as "other" with a note appended as "post roman
glazed". Sidenote--I have no idea what this means. *

ERROR:   Sheet 345: Cell A14 has unknown value rooftiles. -> sheet skipped
*Should be ignored if present.*

ERROR:   Sheet 345: Cell A15 has unknown value bricks. -> sheet skipped
*Can be ignored. Ideally moved to other collected material category of
"CBM" or "Ceramic Building Material". I don't know if this is possible. *

WARNING: Sheet 345: Missing rows
coarseware,stone,invetriata,maiolica,dolia,african rs. -> ignored.
*Not sure about this, seems like the sheet didn't get filled in?*

ERROR:   Sheet 345 cannot be imported.
*Dunno.*

 I suspect "mni" can be ignored as only a few of those sheets actually have
it
*Yes*

 Courseware I would translate into Coarseware
*Yes*

what about the other uknown values?
*I tried to answer above, not entirely sure. Ignore the sheet and I can
follow it up.*

what about the missing values except from coarseware?
*I tried to answer above, not entirely sure. Ignore the sheet and I can
follow it up.*

Will respond to others shortly.

best,
Tyler
---

Tyler Franconi
Visiting Assistant Professor of Archaeology
Joukowsky Institute for Archaeology & the Ancient World
Brown University
Box 1837, 60 George Street, Providence, RI  02912, USA
***@***.***
